Internal error: Cannot read properties of undefined (reading 'tableName')

Seeing quite a few of these errors:

```
/Users/x/Developer/contra/gaia/apps/contra-api/src/temporal-workflows/routines/invoice/sendInvoiceOverdueFiveDaysReminderEmail.ts
  20:43  error  Internal error: Cannot read properties of undefined (reading 'tableName')  @ts-safeql/check-sql

/Users/x/Developer/contra/gaia/apps/contra-api/src/temporal-workflows/routines/invoice/sendInvoiceOverdueTwoDaysReminderEmail.ts
  23:43  error  Internal error: Cannot read properties of undefined (reading 'tableName')  @ts-safeql/check-sql
```

Looking at the underlying queries, nothing really stands out as unusual, e.g.

```ts
const result = await pool.one(sql.type(InvoiceDetailsShape)`
  SELECT
    pp.client_user_account_id,
    contractor.id contractor_user_account_id,
    contractor.given_name contractor_first_name,
    client.id client_user_account_id,
    client.given_name client_first_name,
    i.total_amount invoice_total_amount
  FROM invoice i
  INNER JOIN paid_project pp ON
    pp.id = i.paid_project_id
  INNER JOIN user_account client ON
    client.id = pp.client_user_account_id
  INNER JOIN user_account contractor ON
    contractor.id = pp.contractor_user_account_id
  LEFT JOIN workspace_team wt ON wt.id = pp.client_workspace_team_id
  WHERE
    i.id = ${invoiceId}
`);
```

--- a/packages/generate/src/generate.test.ts
+++ b/packages/generate/src/generate.test.ts
@@ -1303,3 +1303,16 @@ test("should distinguish between schema", async () => {
     ],
   });
 });
+
+test("select with duplicate columns and alias", async () => {
+  await testQuery({
+    query: `
+      SELECT
+        caregiver.id as x,
+        caregiver_certification.caregiver_id as x
+      FROM caregiver
+        JOIN caregiver_certification ON caregiver.id = caregiver_certification.caregiver_id
+    `,
+    expectedError: `Duplicate columns: caregiver.id (alias: x), caregiver_certification.caregiver_id (alias: x)`,
+  });
+});

--- a/packages/generate/src/generate.ts
+++ b/packages/generate/src/generate.ts
@@ -218,15 +218,28 @@ async function generate(
     );
 
     if (duplicateCols.length > 0) {
-      const dupes = duplicateCols.map((col) => ({
-        table: pgColsByTableOidCache.get(col.table)!.find((c) => c.colName === col.name)!.tableName,
-        column: col.name,
-      }));
+      const dupes = duplicateCols.map((col) => {
+        const tableCols = pgColsByTableOidCache.get(col.table);
+        const tableCol =
+          col.number !== 0
+            ? tableCols?.at(col.number - 1)
+            : tableCols?.find((c) => c.colName === col.name);
+
+        return {
+          table: tableCol?.tableName,
+          column: col.name,
+          originalColumn: tableCol?.colName,
+        };
+      });
 
       return either.left(
         DuplicateColumnsError.of({
           queryText: query,
-          columns: dupes.map((x) => `${x.table}.${x.column}`),
+          columns: dupes.map((x) => {
+            return x.column !== x.originalColumn
+              ? `${x.table}.${x.originalColumn} (alias: ${x.column})`
+              : `${x.table}.${x.column}`;
+          }),
         }),
       );
     }

